### ISSUE TITLE
Error response from daemon: create foobar: create foobar: Error looking up volume plugin lvm: Plugin.Activate: 400 Bad Request: malformed Host header

### ISSUE THREAD

**Issue by xianhuawei** [projectatomic/docker-lvm-plugin#8]
Error response from daemon: create foobar: create foobar: Error looking up volume plugin lvm: Plugin.Activate: 400 Bad Request: malformed Host header

**Comment by shishir-a412ed** [comment]
@xianhuawei Can you give more details ?

1) Exact steps you did to produce this error.
2) Output of `docker version`.
3) Output of `docker info`.

**Comment by xianhuawei** [comment]
go get github.com/projectatomic/docker-lvm-plugin
cd docker-lvm-plugin
make
sudo make install


docker version
Client:
 Version:         1.10.3
 API version:     1.22
 Package version: docker-common-1.10.3-44.el7.centos.x86_64
 Go version:      go1.4.2
 Git commit:      9419b24-unsupported
 Built:           Fri Jun 24 12:09:49 2016
 OS/Arch:         linux/amd64

Server:
 Version:         1.10.3
 API version:     1.22
 Package version: docker-common-1.10.3-44.el7.centos.x86_64
 Go version:      go1.4.2
 Git commit:      9419b24-unsupported
 Built:           Fri Jun 24 12:09:49 2016
 OS/Arch:         linux/amd64




docker info

Containers: 11
 Running: 1
 Paused: 0
 Stopped: 10
Images: 49
Server Version: 1.10.3
Storage Driver: devicemapper
 Pool Name: docker-253:0-68523079-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.822 GB
 Data Space Total: 107.4 GB
 Data Space Available: 33.22 GB
 Metadata Space Used: 5.841 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.142 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2015-12-01)
Execution Driver: native-0.2
Logging Driver: journald
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 3.10.0-229.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 2
CPUs: 8
Total Memory: 7.546 GiB
Name: localhost.localdomain
ID: JEN3:7WRN:MRUU:CIH3:3ICR:B2VL:S4BF:PG2S:7OGH:SC3A:6UYM:P6SG
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Registries: docker.io (secure)

**Comment by xianhuawei** [comment]
ps aux|grep docker

root      4555  0.0  0.0 113844  4120 ?        Ssl  Jul04   0:00 /usr/libexec/docker/docker-lvm-plugin
root      4882  0.0  0.0 115244  1424 ?        Ss   Jul04   0:00 /bin/sh -c /usr/bin/docker-current daemon            --exec-opt native.cgroupdriver=systemd            $OPTIONS            $DOCKER_STORAGE_OPTIONS            $DOCKER_NETWORK_OPTIONS            $ADD_REGISTRY            $BLOCK_REGISTRY            $INSECURE_REGISTRY            2>&1 | /usr/bin/forward-journald -tag docker
root      4883  1.1  0.5 865916 42096 ?        Sl   Jul04  13:09 /usr/bin/docker-current daemon --exec-opt native.cgroupdriver=systemd --selinux-enabled --log-driver=journald
root      4884  0.0  0.0 101728  4168 ?        Sl   Jul04   0:00 /usr/bin/forward-journald -tag docker
root     23959  0.0  0.0 112648   972 pts/0    S+   10:17   0:00 grep --color=auto docker

**Comment by xianhuawei** [comment]
cat /etc/centos-release
CentOS Linux release 7.1.1503 (Core) 


yum install -y docker

**Comment by shishir-a412ed** [comment]
@xianhuawei and I am assuming you updated your volume group name in `/etc/docker/docker-lvm-plugin` ? before creating the logical volume using this plugin.

**Comment by xianhuawei** [comment]
yes 

[root@localhost ~]# vgdisplay |grep group_one
  VG Name               volume_group_one
[root@localhost ~]# cat /etc/docker/docker-lvm-plugin 
# Update volume group (vg) name in this file. A volume group name is needed 
# to create logical volumes using docker lvm plugin. You can choose an existing
# volume group name by listing volume groups on your system using `vgs` command 
# OR create a new volume group using `vgcreate` command. 
# e.g. vgcreate volume_group_one /dev/hda where /dev/hda is your partition or
# whole disk on which physical volumes were created.
VOLUME_GROUP=volume_group_one

**Comment by cschauer** [comment]
I ran into this as well. It's an incompatibility with golang versions. Docker on CentOS/RHEL is compiled with go1.4.2. There were some changes to the http library in go1.6 that requires the Host header in requests. You need to build the plugin with go1.5 or lower.

**Comment by xianhuawei** [comment]
@cschauer good!
@shishir-a412ed can fix it?

**Comment by cschauer** [comment]
@xianhuawei Try installing from the package:  `yum install docker-lvm-plugin`

**Comment by shishir-a412ed** [comment]
@cschauer @xianhuawei 
I am able to run `docker-lvm-plugin` successfully i.e create a volume using the plugin, bind mount it inside the container etc. I don't face this error.

Now I work on a fedora 23 machine, so I had to build it from the source.
my go version is:
```
go version go1.6 linux/amd64
```

I think I faced this error in the past, but this was related to systemd unit files. My systemd unit files were old. once I updated them to a more current version, these errors seem to have disappeared.

Shishir
